plugin/keyring_vault/vault_base64.h

```diff
@@ -0,0 +1,23 @@
+#ifndef MYSQL_VAULT_BASE64_H
+#define MYSQL_VAULT_BASE64_H
+
+#include <my_global.h>
+#include "vault_memory.h"
+
+namespace keyring
+{
+  class Vault_base64
+  {
+  public :
+    enum Base64Format
+    {
+      SINGLE_LINE,
+      MULTI_LINE
+    };
+    static bool encode(const void *src, size_t src_len, Secure_string *encoded, Base64Format format);
+    static bool decode(const Secure_string &src, char **dst, uint64 *dst_length);
```

Comment:
It needs to be uint64, this is what is required by base64.h base64_needed_decoded_length. The second function is used by me in vault_parser, where I need memory from a buffer, so to not double allocate the buffer I use this function. I have added a comment that memory allocated by this function should be deallocated with delete[].